File: src/lib/supabase.ts
import { createClient } from '@supabase/supabase-js';
import { env } from '../env/client';

export const supabase = env.VITE_SUPABASE_URL && env.VITE_SUPABASE_ANON
  ? createClient(env.VITE_SUPABASE_URL, env.VITE_SUPABASE_ANON)
  : null;

File: backend/app/core/supabase.py
from supabase import Client, create_client

from backend.app.core.config import settings


def get_supabase() -> Client | None:
    if not settings.SB_URL or not settings.SB_SECRET_KEY:
        return None
    try:
        return create_client(settings.SB_URL, settings.SB_SECRET_KEY)
    except Exception:
        return None


def get_supabase_client() -> Client | None:
    return get_supabase()


supabase = get_supabase()
